pkg/runtime/k0s/init.go

```diff
@@ -36,46 +42,100 @@ func (k *Runtime) init() error {
 	return nil
 }
 
-// MergeConfigOnMaster0 convert the cluster file spec to kubectl.yaml (k0sctl config file) to lead cluster run
-func (k *Runtime) MergeConfigOnMaster0() error {
-	if err := k.K0sConfig.ConvertTok0sConfig(k.cluster); err != nil {
-		return fmt.Errorf("failed to convert to k0s config from clusterfile: %v", err)
+// GenerateConfigOnMaster0 generate the k0s.yaml to /etc/k0s/k0s.yaml and lead the controller node install
+func (k *Runtime) GenerateConfigOnMaster0() error {
+	if err := k.generateK0sConfig(); err != nil {
+		return fmt.Errorf("failed to generate config: %v", err)
+	}
+	k.modifyConfigRepo()
+	if err := k.marshalToFile(); err != nil {
+		return fmt.Errorf("failed to write k0s config: %v", err)
 	}
-	ssh, err := k.getHostSSHClient(k.cluster.GetMaster0IP())
+	return nil
+}
+
+// GenerateCert generate the containerd CA for registry TLS and k0s join token for 100 years.
+func (k *Runtime) GenerateCert() error {
+	if err := k.generateK0sToken(); err != nil {
+		return err
+	}
+
+	if err := k.GenerateRegistryCert(); err != nil {
+		return err
+	}
+	return k.SendRegistryCert(k.cluster.GetMasterIPList()[:1])
+}
+
+func (k *Runtime) generateK0sConfig() error {
+	master0IP := k.cluster.GetMaster0IP()
+	ssh, err := k.getHostSSHClient(master0IP)
 	if err != nil {
-		return fmt.Errorf("failed to get ssh client: %v", err)
+		return err
 	}
-	output, err := ssh.Cmd(k.cluster.GetMaster0IP(), GetK0sVersionCMD)
+	createCMD := "mkdir -p /etc/k0s && k0s config create"
```

Comment:
could we use k0s client code to generate it Or load default config file from rootfs, rather than shell command to gen k0sconfig file ?

Reply:
Have an other way to generate it.

Reply:
Please separate the two single commands, like:
```
mkdirCmd := "mkdir -p /etc/k0s"
k0sConfigCreateCmd := "k0s config create" 
```
 Otherwise the golang code would be quite coupled with the command bundles.